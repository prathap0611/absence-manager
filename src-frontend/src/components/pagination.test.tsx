import { render, fireEvent } from '@testing-library/react';
import Pagination from './pagination';

describe('Pagination Tests', () => {
    const setCurrentPage = jest.fn();

    beforeEach(() => {
        setCurrentPage.mockReset();
    })

    it('should render the page correctly', () => {
        const {asFragment} = render(
            <Pagination
                totalNumberOfPages={5}
                currentPage={0}
                setCurrentPage={setCurrentPage}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('should set prev page', () => {
        const { getByTestId } = render(
            <Pagination
                totalNumberOfPages={5}
                currentPage={1}
                setCurrentPage={setCurrentPage}
            />
        );
        fireEvent.click(getByTestId('prev-page'));
        expect(setCurrentPage).toBeCalledWith(0);
    });

    it('should disable prev page when current page number is 0', () => {
        const { getByTestId } = render(
            <Pagination
                totalNumberOfPages={5}
                currentPage={0}
                setCurrentPage={setCurrentPage}
            />
        );
        const prevButton = getByTestId('prev-page')
        expect(prevButton).toBeDisabled();

        fireEvent.click(prevButton);

        expect(setCurrentPage).toBeCalledTimes(0);
    });

    it('should set next page', () => {
        const { getByTestId } = render(
            <Pagination
                totalNumberOfPages={5}
                currentPage={1}
                setCurrentPage={setCurrentPage}
            />
        );
        const nextButton = getByTestId('next-page')
        fireEvent.click(nextButton);

        expect(setCurrentPage).toBeCalledWith(2);
    });


    it('should disabled next page when current page is one less than total number of pages', () => {
        const { getByTestId } = render(
            <Pagination
                totalNumberOfPages={5}
                currentPage={4}
                setCurrentPage={setCurrentPage}
            />
        );
        const nextButton = getByTestId('next-page')
        expect(nextButton).toBeDisabled();
        fireEvent.click(nextButton);

        expect(setCurrentPage).toBeCalledTimes(0);
    });
});
