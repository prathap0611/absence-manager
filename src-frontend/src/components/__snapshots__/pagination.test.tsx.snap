// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Tests should render the page correctly 1`] = `
<DocumentFragment>
  <section
    class="pagination-group"
  >
    <button
      aria-label="previous page"
      class="page-btn"
      data-testid="prev-page"
      disabled=""
    >
      ◀
    </button>
    <span
      class="page-info"
    >
      1 of 5
    </span>
    <button
      aria-label="next page"
      class="page-btn"
      data-testid="next-page"
    >
      ▶
    </button>
  </section>
</DocumentFragment>
`;
